### PR TITLE
Use tokens for primary auth instead of Shibboleth

### PR DESCRIPTION
**Pull request by nwalters512**
This replaces our use of Shibboleth as the primary authentication mechanism with JSON Web Tokens. We still use Shib to get a user's identity, but we create a JWT with that identity, store it in a cookie, and use it for future requests. The benefits of this are outlined in the high-level docs included in this PR.

Before this goes into production, we need to do the following:
* Remove Shib protection from `/q`
* Add Shib protection specifically to `/q/auth/shib`
* Generate and store a secure secret in the `JWT_SECRET` environment variable

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -49,21 +68,30 @@ class Header extends React.Component {
...
+                  <NavLink className="navbar-text mr-3">
+                    <FontAwesomeIcon icon={faUser} className="mr-2" />
+                    {userName}
+                  </NavLink>
+                </Link>
+                <NavItem>
+                  <Button
+                    color="secondary"
+                    type="button"
+                    onClick={() => { window.location = logoutRoute}}
```
Comment (james9909) [inline src/components/Header.js]: I think we should use `Router.pushRouter` here to keep things consistent, unless there's a reason for using `window.location`.

Comment (nwalters512) [reply src/components/Header.js]: I'm generally trying to keep `Router.*` reserved for things that are actually routable client-side - when logging out, we can't do that.

Comment (nwalters512) [reply src/components/Header.js]: Although, an interesting option would be hitting the logout link in the background so we don't need to land the user on the Shib logout page... thoughts?

File: src/components/DevModeLogin.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  handleNetidChange(event) {
+    this.setState({
+      netid: event.target.value,
+    })
+  }
+
+  handleSubmit() {
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply src/components/DevModeLogin.js]: Good catch!

File: src/components/Header.js (hunk 1 above)
Comment (james9909) [reply src/components/Header.js]: Makes sense, I think either way is fine.

**Pull request merged**